ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/editor/editorconfig/TextEditorConfiguration.java

```diff
@@ -97,4 +98,11 @@
      */
     @Nullable
     ChangeInterceptorProvider getChangeInterceptorProvider();
+
+    /**
+     * Return the {@link SignatureHelpProvider}
+     * @return the signature help provider
+     */
+    @Nullable
+    SignatureHelpProvider getSignatureHelpProvider();
```

Comment:
Maybe, but we need to replace all other methods in this class to return `Optional<>` which will be very big changes in  Editor API. In future we definitely should avoid returning `null` and use `Optional<>`